pkg/trie/triedb/lookup.go

```diff
@@ -103,14 +107,29 @@ func (l *TrieLookup) lookupNode(keyNibbles []byte) (codec.Node, error) {
 	}
 }
 
-func (l *TrieLookup) lookupValue(keyNibbles []byte) ([]byte, error) {
+func (l *TrieLookup) lookupValue(keyNibbles []byte) (value []byte, err error) {
+	if l.cache != nil {
+		if value = l.cache.GetValue(keyNibbles); value != nil {
+			return value, nil
+		}
+	}
+
 	node, err := l.lookupNode(keyNibbles)
```

Comment:
While looking up the node do you want to cache it as well?

Reply:
Done! thanks for raising that